numba/core/types/npytypes.py

```diff
@@ -430,6 +434,10 @@ def __init__(self, dtype, ndim, layout, readonly=False, name=None,
         if (not aligned or
             (isinstance(dtype, Record) and not dtype.aligned)):
             self.aligned = False
+        if isinstance(dtype, NestedArray):
+            tmp = Array(dtype.dtype, dtype.ndim, 'C')
```

Comment:
is `tmp` necessary? since the `Array` constructor does not transform the inputs it should be possible to do 
```python
            ndim += dtype.ndim
            dtype = dtype.dtype
```